covalent/_shared_files/util_classes.py

```diff
@@ -19,34 +19,55 @@
 # Relief from the License may be granted by purchasing a commercial license.
 
 
-from dataclasses import dataclass
-from typing import NamedTuple
+import asyncio
+from typing import Any, NamedTuple
 
+from . import logger
 
-@dataclass
-class Status:
-    STATUS: str
+app_log = logger.app_log
+log_stack_info = logger.log_stack_info
 
-    def __bool__(self):
-        """
-        Return True if the status is not "NEW_OBJECT"
-        """
 
-        return self.STATUS != "NEW_OBJECT"
+class SafeVariable(asyncio.Queue):
+    def __init__(self, event_loop=None) -> None:
+        self.event_loop = event_loop or asyncio.get_event_loop()
+        super().__init__(maxsize=1)
 
-    def __str__(self) -> str:
-        return self.STATUS
+    def put_nowait_safe(self, value: Any) -> None:
+        self.event_loop.call_soon_threadsafe(self.put_nowait, value)
+
+    def get_nowait_safe(self) -> Any:
+        return self.event_loop.call_soon_threadsafe(self.get_nowait, ())
 
+    def save(self, value: Any) -> None:
+        try:
+            self.put_nowait_safe(value)
+        except asyncio.QueueFull:
+            self.get_nowait_safe()
```

Comment:
I'm a bit worried about a race condition here. Let the dispatcher be running on thread 0 and the (non-async) `executor.execute()` be running in thread 1. 
Consider the following scenario:

0) SafeVariable is already set. The status listener in Thread 0 calls `SafeVariable.get()`.
1) In `SafeVariable.get()`, [line 155](https://github.com/python/cpython/blob/455b494386678f7f20c0cd43407e689f14afac15/Lib/asyncio/queues.py#L155) `if self.empty()`is evaluated; the queue is not empty.
2) Thread 0 is preempted by the OS; Thread 1 starts running.
3) Thread 1 calls `SafeVariable.save()`. Since the queue is nonempty, the `QueueFull` exception is raised, and `self.get_nowait_safe()` pops the value from the queue. 
4) Control switches back to Thread 0. Since `self.empty()` already evaluated to False in step 1, the interpreter jumps to [line 174](https://github.com/python/cpython/blob/455b494386678f7f20c0cd43407e689f14afac15/Lib/asyncio/queues.py#L174) `return self.get_noawait()`. This raises an unhandled `asyncio.QueueEmpty` exception.

Reply:
After our discussion, it seems the better way would be to make this back to `queue.Queue` to avoid race conditions, and solve the problem of frequent calls to `get` inside the listener loop.